controllers/api.controllers.js

```diff
@@ -10,11 +11,28 @@ const getTopics = (req, res, next) => {
 const getApiEndpoints = (req, res) => {
     selectApiEndPoints(req.query)
     .then((endpoints) => {
-        console.log(endpoints)
         res.status(200).send({endpoints})
     })
 }
 
+const getArticleById = (req, res) => {
+    let id = req.params.article_id
+    selectArticleById(id)
+    .then((rows) => {
+        if (rows.length === 0) {
+            res.status(404).send({msg: "Not found"})
+        } else {
+            res.status(200).send(rows[0])
+        }
```

Comment:
I'd recommend returning the rows[0] in your models instead. Returning the entire rows array in your controller on a key future-proofs your code in case you need to change or expand its functionality in the future. 🤔 If this way works for you that's fine but remember to keep it consistent throughout 👍  The model should be in charge of all data manipulation.